cleanlab/token_classification/rank.py

```diff
@@ -0,0 +1,112 @@
+
+    sentence_scoring_method: {"min", "softmin"}, default="min"
+        sentence scoring method to aggregate token scores.
+
+        - `min`: sentence score = minimum token label score of the sentence
+        - `softmin`: sentence score = <s, softmax(1-s, t)>, where s denotes the token label scores of the sentence,
+        and <a, b> == np.dot(a, b). The parameter `t` controls parameter of softmax, such that when t -> 0, the
+        method approaches to `min`. This method is the "softer" version of `min`, which adds some minor weights to
+        other scores.
+
+    token_scoring_method: {"self_confidence", "normalized_margin", "confidence_weighted_entropy"}, default="self_confidence"
+        label quality scoring method. See `cleanlab.rank.get_label_quality_scores` for more info.
+
+    param: float, default=0.04
+        temperature of softmax. If sentence_scoring_method == "min", `param` is ignored.
+
+    tokens: list, optinal, default=None
+        tokens in nested list format, such that `tokens[i]` is a list of tokens for the i'th sentence. See return value
+        `token_info` for more info.
+
+    return_scores_per_token: bool, default=True
+        If set to True, returns additional token information. See return value `token_info` for more info.
+
+    Returns
+    ----------
+    sentence_scores: np.array
+        A vector of sentence scores between 0 and 1, where lower scores indicate sentence is more likely to contain at
+        least one label issue.
+
+    token_info: list
+        Returns only if `return_scores_per_token=True`. A list of pandas.Series, such that token_info[i] contains the
+        token scores for the i'th sentence. If tokens are provided, the series is indexed by the tokens.
+
+    ----------
+    """
+    methods = ["min", "softmin"]
+    assert sentence_scoring_method in methods, "Select from the following methods:\n%s" % "\n".join(
+        methods
+    )
+
+    labels_flatten = np.array([l for label in labels for l in label])
+    pred_probs_flatten = np.array([p for pred_prob in pred_probs for p in pred_prob])
+    n, m = pred_probs_flatten.shape
+
+    # TODO: maybe move this to a helper in the module? may be helpful for users
+    sentence_length = [len(label) for label in labels]
+
+    def nested_list(x, length):
+        i = iter(x)
+        return [[next(i) for _ in range(length)] for length in sentence_length]
+
+    token_scores = main_get_label_quality_scores(
+        labels=labels_flatten, pred_probs=pred_probs_flatten, method=token_scoring_method
+    )
+    scores_nl = nested_list(token_scores, sentence_length)
+
+    if sentence_scoring_method == "min":
+        fun = lambda scores: np.min(scores)
+
+    elif sentence_scoring_method == "softmin":
+        softmax = lambda scores: np.exp(scores / param) / np.sum(np.exp(scores / param))
+        fun = lambda scores: np.dot(scores, softmax(1 - np.array(scores)))
```

Comment:
Let's make this a separate helper function:

```
def _softmin_sentence_score(token_scores, param=0.04, **kwargs):
    softmax = lambda scores: np.exp(scores / param) / np.sum(np.exp(scores / param))
    fun = lambda scores: np.dot(scores, softmax(1 - np.array(scores)))
```
Note here kwargs is simply ignored, but it exists so other args could be passed in without causing this function to crash.